sdk/search/Azure.Search.Documents/src/Indexes/Models/LexicalNormalizerName.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Search.Documents.Indexes.Models
+{
+    public readonly partial struct LexicalNormalizerName
+    {
+#pragma warning disable CA1034 // Nested types should not be visible
+        /// <summary>
+        /// The values of all declared <see cref="LexicalNormalizerName"/> properties as string constants.
+        /// These can be used in <see cref="SimpleFieldAttribute"/>, <see cref="SearchableFieldAttribute"/> and anywhere else constants are required.
+        /// </summary>
+        public static class Values
```

Comment:
This is an interesting pattern.   Can you help me understand what the intention is for it?

Reply:
We are using `LexicalNormalizerName` constant values for `SimpleFieldAttribute` and `SearchableFieldAttribute`. You can see an example [here](https://github.com/Azure/azure-sdk-for-net/blob/5293a0c02298c3ad359962a0caf2395c2bf56986/sdk/search/Azure.Search.Documents/tests/Models/ReflectableModel.cs#L96).

I tried using an [existing enum type](https://github.com/Azure/azure-sdk-for-net/blob/5293a0c02298c3ad359962a0caf2395c2bf56986/sdk/search/Azure.Search.Documents/src/Generated/Models/LexicalNormalizerName.cs#L14) for the attributes [here](https://github.com/Azure/azure-sdk-for-net/blob/5293a0c02298c3ad359962a0caf2395c2bf56986/sdk/search/Azure.Search.Documents/src/Indexes/SimpleFieldAttribute.cs#L58) but compiler gives `LexicalNormalizerName is not a valid attribute parameter type` error:

![image](https://user-images.githubusercontent.com/45216704/176740119-6eb37e26-b6e7-4bbd-9670-99f047efe976.png)

So I used the existing pattern for `LexicalNormalizerName` that we using for [LexicalAnalyzerName](https://github.com/Azure/azure-sdk-for-net/blob/5293a0c02298c3ad359962a0caf2395c2bf56986/sdk/search/Azure.Search.Documents/src/Indexes/Models/LexicalAnalyzerName.cs#L13) and it works well for attributes.

Reply:
I was poking more at "why are we wrapping values in an inner static class rather than just exposing the members".

Reply:
>@jsquire - why are we wrapping values in an inner static class rather than just exposing the members

Because those members already exist in the generated type [here](https://github.com/Azure/azure-sdk-for-net/blob/5293a0c02298c3ad359962a0caf2395c2bf56986/sdk/search/Azure.Search.Documents/src/Generated/Models/LexicalNormalizerName.cs#L32-L40).

So compiler gives this error:
![image](https://user-images.githubusercontent.com/45216704/176755620-c823f9d4-41c9-4e61-9e2c-2d3b5a32a9aa.png)

Reply:
This was the suggested work around from @tg-msft and/or @KrzysztofCwalina at the time since, yes, you can't use the "extensible enum" struct as an attribute value (not constant).